bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/Workspace.java

```diff
@@ -2277,8 +2276,12 @@ private void setExplicitWorkspaceEncoding() {
 		// Therefore check preferences directly (encoding could be set even in a new
 		// workspace via product customization file, like
 		// org.eclipse.core.resources/encoding=ISO-8859-1)
-		String defaultEncoding = Platform.getPreferencesService().getString(ResourcesPlugin.PI_RESOURCES,
-				ResourcesPlugin.PREF_ENCODING, /* default */null, /* all scopes */null);
+		// Using the (deprecated) Preferences API here because otherwise we created
```

Comment:
I **believe** (have only tried in debugger) that simply moving `initializePreferenceLookupOrder();` **after** this call in `open()` should solve the problem

Reply:
As soon as one calls `initializePreferenceLookupOrder()` code using the preference API could (indirectly) access the not fully opened Workspace.

Reply:
Could you please elaborate how is that possible?

Reply:
Lets just keep it as is as described below, no need to complicate things here, I think its rather theoretical